python/ShapeTools.py

```diff
@@ -65,16 +69,63 @@ def doIndividualModels(self):
                 pdf.setStringAttribute("combine.process", p)
                 pdf.setStringAttribute("combine.channel", b)
                 pdf.setAttribute("combine.signal", self.DC.isSignal[p])
+                if self.DC.isSignal[p]:
+                    pdf.setAttribute('skipForErrorSum')
                 coeff.setStringAttribute("combine.process", p)
                 coeff.setStringAttribute("combine.channel", b)
                 coeff.setAttribute("combine.signal", self.DC.isSignal[p])
                 pdfs.add(pdf); coeffs.add(coeff)
                 if not self.DC.isSignal[p]:
                     bgpdfs.add(pdf); bgcoeffs.add(coeff)
+                else:
+                    sigcoeffs.append(coeff)
             if self.options.verbose > 1: print "Creating RooAddPdf %s with %s elements" % ("pdf_bin"+b, coeffs.getSize())
-            sum_s = ROOT.RooAddPdf("pdf_bin%s"       % b, "",   pdfs,   coeffs)
+            if self.options.newHist >= 1:
+                prop = ROOT.CMSHistErrorPropagator("prop_bin%s" % b, "", self.out.binVar, pdfs, coeffs)
+                if self.options.newHistBinPars >= 0.:
+                    bbb_args = prop.setupBinPars(self.options.newHistBinPars)
```

Comment:
setupBinPars will create all the new nuisance parameters, but it's up to us to define the constraint pdfs. Attributes will have been set on the parameters that tell us what type of constraints to make.